roles/matrix-base/defaults/main.yml

```diff
@@ -91,7 +91,7 @@ matrix_homeserver_url: "https://{{ matrix_server_fqn_matrix }}"
 # Specifies where the homeserver is on the container network.
 # Where this is depends on whether there's a reverse-proxy in front of it, etc.
 # This likely gets overriden elsewhere.
-matrix_homeserver_container_url: "http://matrix-synapse:8008"
+matrix_homeserver_container_url: "http://matrix-synapse:{{ matrix_synapse_container_client_api_port }}"
```

Comment:
This uses a `matrix-synapse` variable in the `matrix-base` role. We should probably define `matrix_homeserver_container_url` as empty in here and redefine it in `group_vars/matrix_servers` (we're already doing this). We can introduce some `roles/matrix-base/tasks/validate_config.yml` task file, which checks whether `matrix_homeserver_container_url` is empty or not.

Reply:
Ok will leave `matrix_homeserver_container_url` empty in `matrix-base` and redefine it in `group_vars/matrix_servers`